src/dbnode/client/reader_slice_of_slices_iterator.go

```diff
@@ -204,3 +204,7 @@ func (it *readerSliceOfSlicesIterator) Size() (int, error) {
 	}
 	return size, nil
 }
+
+func (it *readerSliceOfSlicesIterator) Rewind() {
+	it.idx = 0
```

Comment:
Looking at `Reset(...)` this should be -1?

Perhaps best to create an internal method called `resetIterationState()` and call that both from `Reset(...)` and `Rewind(...)` that sets it to -1?

Also might be worth writing a test since we didn't catch that previous to this review and would be good to regression test against.

Reply:
+1

Reply:
FYI so the issue here is that via `ResetSliceOfSlices` these multiReaderIterators get pushed forward into a state where idx is zero so that the first `Current` call is valid. Given this, this actually needs to be zero to work properly. However I agree it is confusing instead after discussing w/ Artem I changed to have `Rewind` set to idx -1 but then when we call this we do:
```
		// Rewind the reader state back to beginning to the it can be re-iterated by caller.
		// These multi-readers are queued up via ResetSliceOfSlices so that the first Current
		// index is set, and therefore we must also call an initial Next here to match that state.
		// This behavior is not obvious so we should later change ResetSliceOfSlices to not do this
		// initial Next move and assert that all iters start w/ Current as nil.
		readers.Rewind()
		readers.Next()
```
w/ a comment describing how this behavior probably should change.